employee-rostering-backend/src/main/java/org/optaweb/employeerostering/service/SkillService.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.employeerostering.service;
+
+import java.util.List;
+
+import javax.persistence.EntityNotFoundException;
+
+import org.optaweb.employeerostering.domain.Skill;
+import org.optaweb.employeerostering.persistence.SkillRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.util.Assert;
+
+@Service
+public class SkillService {
+
+    private final SkillRepository skillRepository;
+
+    public SkillService(SkillRepository skillRepository) {
+        this.skillRepository = skillRepository;
+        Assert.notNull(skillRepository, "skillRepository must not be null.");
```

Comment:
This null check is kind of redundant. If the Spring IoC cannot find a bean satisfying the `skillRepository` dependency it will [fail](https://www.baeldung.com/spring-nosuchbeandefinitionexception) earlier, so this is an unreachable statement.